src/client/app/service/google-realtime/google-realtime.service.ts

```diff
@@ -0,0 +1,230 @@
+import {Injectable} from '@angular/core';
+import {
+    Http, URLSearchParams, Response,
+    RequestOptions, Headers, QueryEncoder
+} from '@angular/http';
+import {AsyncSubject, Observable} from 'rxjs';
+import CollaborativeString = gapi.drive.realtime.CollaborativeString;
+
+
+const API_KEY = 'AIzaSyBcALBUoAgCQ--XxyHjIWW6ifBEyDSck08';
+const CLIENT_ID =
+    '458941249796-jfvnbelhroiit38vhe5d69av0jjnoi7b.apps.googleusercontent.com';
+
+
+const GOOGLE_APIS_FILES_URL = 'https://www.googleapis.com/drive/v3/files';
+const GOOGLE_DRIVE_FIELDS_TO_QUERY = 'nextPageToken, files(id, name)';
+const PAGE_SIZE = 10;
+
+const UXGRAPH_MIME_TYPE = 'application/vnd.google.drive.ext-type.uxgraph';
+
+
+/**
+ * Represents a single Google Drive file.
+ *
+ * This object is just metadata for the file. Data for the file is stored in
+ * the Realtime Document.
+ *
+ * TODO(girum): Create an ngrx/store domain layer that automatically implements
+ * Realtime Document manipulation.
+ */
+export interface DriveFile {
+  id: string;
+  name: string;
+  mimeType: string;
+  kind: string;
+}
+
+/**
+ * A service to authenticate and interact with Google's Realtime API.
+ * Includes Google Drive API interaction.
+ */
+@Injectable()
+export class GoogleRealtimeService {
+
+  /**
+   * The OAuth token that Google gives back for the current client.
+   * TODO(girum): This should be a BehaviorSubject instead to handle reconnects.
+   *
+   * @type {AsyncSubject<GoogleApiOAuth2TokenObject>}
+   */
+  oauthToken: AsyncSubject<GoogleApiOAuth2TokenObject> =
+      new AsyncSubject<GoogleApiOAuth2TokenObject>();
+
+  constructor(private http: Http) {
+    // Immediately load additional JavaScript code to interact with gapi
+    // (gapi = "Google API" for JS).
+    gapi.load('auth:client,drive-realtime,drive-share', () => {
```

Comment:
Yeah it's defined as a global in a d.ts file:
https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/gapi